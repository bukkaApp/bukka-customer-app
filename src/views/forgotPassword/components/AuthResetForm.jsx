import React from 'react';

import PropTypes from 'prop-types';

import Container from 'Components/container';

import TextField from '../common/TextField';
import AuthButton from '../common/AuthButton';
import CreateAccount from '../common/CreateAccount';

import './authresetform.scss';

const ErrorMessage = ({ message }) => {
  if (message) {
    return (
      <div className="text-danger help-block text-center my-4">{message}</div>
    );
  }
  return null;
};

const preventDefault = event => event.preventDefault();

const AuthResetForm = ({
  push,
  instruction,
  handleChange,
  handleSubmit,
  domStructure,
  validationErrors,
  isFormCompleted,
  errorMessage,
  title,
}) => (
  <>
    <Container>
      <h2 className="text-center pt-4 mb-0 auth-title">Reset password</h2>
      <form
        className="pb-3 form-auth"
        name="form"
        onSubmit={!isFormCompleted ? preventDefault : handleSubmit}
      >
        <div className="form-title mb-4">
          <span>{instruction || 'Type your email'}</span>
        </div>
        <div className="col-12 mt-9">
          <TextField
            handleChange={handleChange}
            domStructure={domStructure}
            validationErrors={validationErrors}
          />
        </div>
        <ErrorMessage message={errorMessage} />
        <AuthButton
          isFormCompleted={isFormCompleted}
          title={title}
        />
      </form>
    </Container>
    <div className="pb-4">
      <CreateAccount push={push} />
    </div>
  </>
);

export default AuthResetForm;

AuthResetForm.defaultProps = {
  push: () => {},
  domStructure: {},
  isFormCompleted: false,
  title: '',
  validationErrors: {},
  errorMessage: '',
  instruction: '',
};

AuthResetForm.propTypes = {
  push: PropTypes.func,
  instruction: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  domStructure: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
  validationErrors: PropTypes.objectOf(PropTypes.string),
  isFormCompleted: PropTypes.bool,
  title: PropTypes.string,
  errorMessage: PropTypes.string
};

ErrorMessage.defaultProps = {
  message: '',
};

ErrorMessage.propTypes = {
  message: PropTypes.string,
};
