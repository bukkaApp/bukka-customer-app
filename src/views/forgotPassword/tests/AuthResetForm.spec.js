import React from 'react';
import AuthResetForm from '../components/AuthResetForm';
import Data from './__mocks__/register';

describe('AuthResetForm component', () => {
  const props = {
    instruction: 'Login in password',
    title: 'Sign Up',
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    domStructure: Data.domStructureMock,
    validationErrors: Data.validationErrorsMock,
    isFormCompleted: true,
    type: 'signin',
    errorMessage: 'An Error occur'
  };
  const wrapper = shallow(<AuthResetForm {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
