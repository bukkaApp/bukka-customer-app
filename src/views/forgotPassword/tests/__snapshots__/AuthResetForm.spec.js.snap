// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthResetForm component renders properly 1`] = `ShallowWrapper {}`;
